code/firmware/rosco_m68k_firmware/blockdev/bbsd.c

```diff
@@ -197,12 +287,13 @@ bool BBSD_read_block(BBSDCard *sd, uint32_t block, uint8_t *buffer) {
 
     // Read data into buffer
     BBSPI_recv_buffer(buffer, 512);
+    int checksum = ((BBSPI_recv_byte() << 8) | BBSPI_recv_byte()) & 0xFFFF;
 
-    // Get checksum too (and ignore it ;) )
-    BBSPI_recv_byte();
-    BBSPI_recv_byte();
-
-    result = true;
+    // Check the checksum.  Checksum stored big endian on sd card 
+    // Allow for any legacy rosco SD Cards written with a dummy checksum
```

Comment:
I don't think this comment on legacy rosco_m68k cards having been written with a dummy checksum is correct. As far as I know, the CRC is used to protect against over-the-wire errors, not the integrity of blocks on disk, so I don't think that `0xFFFF` being read is related to `0xFFFF` having been written.

Reply:
> You think the card is not storing the crc16 but recalculating and comparing to what’s been sent ?  Do all cards support crc16 ? Is it optional ?  (Or if they don’t support they just ignore?) 

Yes, from the SD Physical Layer Simplified Specification, the CRC used to protect against transmission errors, and it's generated by the card. By default, CRC should be disabled when the card enters SPI mode (with some exceptions, like CMD8), unless it's enabled by using CMD59. With this PR removing the CMD59 that was present, I believe that compliant cards *should* be in CRC OFF mode, but the comment that was there implies that some cards were defaulting to staying in CRC ON without it.

I think that CRC support is a basic feature of SD, and should be present on all cards (it's always enabled in SDIO mode, so the CMD0 to initially switch to SPI must have a valid CRC). The CRC_ON_OFF command (CMD59) is described as a mandatory command to be supported, and is in the "basic" command class.